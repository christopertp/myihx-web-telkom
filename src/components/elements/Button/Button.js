import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function Button(props) {
  const { children, className, disabled, onClick, to, type, variant } = props;
  const customClass = [styles[variant], className].filter(Boolean).join(' ');

  return to ?
    (<Link className={customClass} disabled={disabled} to={to}>{children}</Link>):
    (<button className={customClass} disabled={disabled}
      onClick={onClick} type={type}>{children}</button>);
}

Button.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  onClick: null,
  to: '',
  type: 'button',
  variant: 'red',
};

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  to: PropTypes.string,
  type: PropTypes.string,
  variant: PropTypes.oneOf(['bordered', 'red', 'text', 'white']),
};
