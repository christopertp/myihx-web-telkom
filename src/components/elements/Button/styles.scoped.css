.bordered, .red, .text, .white {
  display: block;
  font-size: 0.9rem;
  font-weight: 800;
  padding: 0.75rem 1.5rem;
  text-align: center;
  text-transform: uppercase;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.red[disabled], .text[disabled] {
  color: rgba(255, 255, 255, 0.5);
  pointer-events: none;
}

.bordered[disabled] {
  color: rgba(51, 52, 53, 0.5);
  pointer-events: none;
}

.bordered {
  background-color: transparent;
  border: 0.0625rem solid var(--primary-color);
  border-radius: 0.5rem;
  color: #333435;
}

.bordered:not(:disabled):hover {
  border-color: #db1e11;
  cursor: pointer;
}

.red {
  background-color: var(--primary-color);
  border: 0.0625rem solid var(--primary-color);
  border-radius: 0.5rem;
  color: #ffffff;
}

.red:not(:disabled):hover {
  background-color: #db1e11;
  border-color: #cf1c10;
  cursor: pointer;
}

.white {
  color: #191919;
  background-color: #fafafa;
  border-color: #fafafa;
  border-radius: 0.5rem;
}
.white:hover {
  background-color: #e7e7e7;
  border-color: #e1e1e1;
}

.text {
  background-color: transparent;
  border: none;
  color: var(--primary-color);
}

.text:not(:disabled):hover {
  cursor: pointer;
}
