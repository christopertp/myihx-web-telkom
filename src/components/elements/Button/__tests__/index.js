import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../Button';

describe('src/components/elements/Button', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Button />);
    expect(tree).toMatchSnapshot();
  });

  test('render link', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Button to="/" />);
    expect(tree).toMatchSnapshot();
  });
});
