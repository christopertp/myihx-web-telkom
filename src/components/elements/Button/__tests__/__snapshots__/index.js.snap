// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Button render 1`] = `
<button
  className=""
  disabled={false}
  onClick={null}
  type="button"
/>
`;

exports[`src/components/elements/Button render link 1`] = `
<Link
  className=""
  disabled={false}
  to="/"
/>
`;
